Jenkinsfile

```diff
@@ -1,252 +1,295 @@
-                    currentTag = getLatestGitTag()
-                    colourText("info", "Found latest tag: ${currentTag}")
-                    newTag =  IncrementTag( currentTag, RELEASE_TYPE )
-                    colourText("info", "Generated new tag: ${newTag}")
-                    // push(newTag, currentTag)
+                    def uploadSpec = """{
+                        "files": [
+                            {
+                                "pattern": "target/universal/*.zip",
+                                "target": "registers-sbt-snapshots/uk/gov/ons/${buildInfo.name}/${buildInfo.number}/"
+                            }
+                        ]
+                    }"""
+                    artServer.upload spec: uploadSpec, buildInfo: buildInfo
                 }
             }
-        }
-
-        stage ('Package and Push Artifact') {
-            agent any
-            when {
-                expression {
-                    env.BRANCH_NAME.toString().equals(BRANCH_PROD)
+            post {
+                success {
+                    colourText("info","Stage: ${env.STAGE_NAME} successful!")
                 }
-            }
-            steps {
-                script {
-                    env.NODE_STAGE = "Package and Push Artifact"
+                failure {
+                    colourText("warn","Stage: ${env.STAGE_NAME} failed!")
                 }
-                sh '''
-                    sbt clean compile package
-                    sbt clean compile assembly
-                '''
-                colourText("success", 'Package.')
             }
-
         }
 
-        stage('Deploy'){
-            agent any
-            when {
-                expression {
-                    env.BRANCH_NAME.toString().equals(BRANCH_DEV) || env.BRANCH_NAME.toString().equals(BRANCH_TEST) || env.BRANCH_NAME.toString().equals(BRANCH_PROD)
-                }
+        stage('Deploy: Dev'){
+            agent { label 'deploy.cf' }
+            when { 
+                branch "master"
+                // evaluate the when condition before entering this stage's agent, if any
+                beforeAgent true 
+            }
+            environment{
+                CREDS = 's_jenkins_sbr_dev'
+                SPACE = 'Dev'
             }
             steps {
                 script {
-                    env.NODE_STAGE = "Deploy"
+                    def downloadSpec = """{
+                        "files": [
+                            {
+                                "pattern": "registers-sbt-snapshots/uk/gov/ons/${buildInfo.name}/${buildInfo.number}/*.zip",
+                                "target": "${distDir}",
+                                "flat": "true"
+                            }
+                        ]
+                    }"""
+                    artServer.download spec: downloadSpec, buildInfo: buildInfo
+                    sh "mv ${distDir}*.zip ${distDir}${env.SVC_NAME}.zip"
```

Comment:
This was required because the pipeline has no knowledge of the build number and I used the wildcard to circumvent solving that in this feature. @ameenhaq has a solution for this, that will remove this.